packages/nx/src/project-graph/project-graph-builder.ts

```diff
@@ -233,7 +233,7 @@ export class ProjectGraphBuilder {
     return this.graph;
   }
 
-  private removeDependenciesWithNode(name: string) {
+  removeDependenciesWithNode(name: string) {
```

Comment:
This was marked public so it can be used in the project-graph-pruning.

There is no harm in exposing this imho. 

Alternative is copy-pasting this code in pruning.